### ISSUE TITLE
Stall on period transition with CEA-608

### ISSUE THREAD

**Issue by sath33sh** [google/shaka-player#2075]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
v2.5.4

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
Yes

**Are you using the demo app or your own custom app?**
Custom app

**If custom app, can you reproduce the issue using our demo app?**
I did not try the demo app

**What browser and OS are you using?**
Chrome 75.0.3770.142 on Mac OS 10.14.5

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What are the manifest and license server URIs?**

https://www.philo.com/manifest/vod/8ee733eef854021a0189826a8dfe1a2c2358bf74/manifest.mpd?content_host=prod.cdn-vdms.philo.com

**What did you do?**

Play a VOD asset with multiple periods and seek from one period to start of next period.

**What did you expect to happen?**
Transition between periods without any errors. 

**What actually happened?**

Playback stalled before the start of next period. Console shows the following assert failures:

```
streaming_engine.js:2207 Assertion failed: All MediaStates should need the same Period or be performing updates.
shaka.media.StreamingEngine.handlePeriodTransition_ @ streaming_engine.js:2207
shaka.media.StreamingEngine.onUpdate_ @ streaming_engine.js:1293
```

I suspect that CEA-608 text stream's mediaState is not updated correctly. Following patch reduces the number of assert failures and fixes the stall, however, presence of assert failures indicate that the patch is not complete. Creating this bug to start a discussion before I debug further.

```
diff --git a/lib/media/streaming_engine.js b/lib/media/streaming_engine.js
index 2b8840cf..e6d04b1d 100644
--- a/lib/media/streaming_engine.js
+++ b/lib/media/streaming_engine.js
@@ -1406,6 +1406,14 @@ shaka.media.StreamingEngine.prototype.update_ = function(mediaState) {
   // of SegmentReferences as an indicator to determine Period boundaries
   // because a SegmentIndex can provide SegmentReferences outside its Period.
   mediaState.needPeriodIndex = needPeriodIndex;
+  if (mediaState.type == ContentType.VIDEO) {
+    // Update needPeriodIndex of text stream if it is CEA closed captions
+    const textState = this.mediaStates_.get(ContentType.TEXT);
+    if (textState && textState.stream.mimeType ==
+          shaka.util.MimeUtils.CLOSED_CAPTION_MIMETYPE) {
+      textState.needPeriodIndex = needPeriodIndex;
+    }
+  }
   if (needPeriodIndex != currentPeriodIndex) {
     shaka.log.debug(logPrefix,
                     'need Period ' + needPeriodIndex,
```

**Comment by joeyparrish** [comment]
Thank you for the report.  I'm in the middle of some changes for v2.6 that will remove periods at the streaming level and push that back into the DASH parser only, so `needPeriodIndex` and those assertions are all on their way out.

How urgently do you need this fixed?  Would the patch above work for you in production until v2.6 is released?  (I don't have a release date for v2.6 yet.)

**Comment by joeyparrish** [comment]
Good news!  This was fixed today and the fix will be in v2.5.5.
